Dockerfile

```diff
@@ -1,17 +1,26 @@
 FROM python:3.12-alpine
 
-# Install python-lxml
-RUN apk add --no-cache --virtual .build-deps \
-    gcc musl-dev \
-    libxslt-dev libxml2-dev && \
-    pip install lxml setuptools && \
-    apk del .build-deps && \
-    apk add --no-cache libxslt libxml2
+RUN adduser -D withings-sync
 
-RUN mkdir -p /src
-COPY . /src
+USER withings-sync
+WORKDIR /home/withings-sync
 
-RUN cd /src && \
-    pip install .
+ENV PATH="/home/withings-sync/.poetry/bin:${PATH}" \
+    PATH="/home/withings-sync/.local/bin:${PATH}" \
+    PIP_ROOT_USER_ACTION=ignore \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    POETRY_HOME=/home/withings-sync/.poetry \
+    POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
 
-ENTRYPOINT ["withings-sync"]
+RUN pip install poetry
+
+COPY --chown=withings-sync:withings-sync pyproject.toml poetry.lock README.md .VERSION ./
+RUN poetry install --without dev --no-root && rm -rf $POETRY_CACHE_DIR
+
+COPY --chown=withings-sync:withings-sync withings_sync ./withings_sync/
+RUN poetry install --without dev
```

Comment:
without a deeper understanding of poetry, can we compbine lines 20 -24 into two lines (copy all the required files and then install it)?